### ISSUE TITLE
Styleguide?

### ISSUE THREAD

**Issue by jcbhmr** [devcontainers-contrib/features#76]
I notice that [some PRs are reformatting things](https://github.com/devcontainers-contrib/features/pull/74/commits/eefddfccd7db50ab16be417c60973d73ab5b978d). I think we should run a code formatted over everything in one big PR to prevent additional superfluous commits in other PRs. Should this be done?

Also, should this repo have a styleguide? Is it large enough to warrant that? If so what format? Anything goes? Use [Prettier](https://prettier.io)? Use [Prettier shell-formatter plugin](https://www.npmjs.com/package/prettier-plugin-sh)?

Tasks
- [x] Prettier/format all files
- [ ] Add a style guide instruction/notice in CONTRIBUTING.md
- [x] Add a GitHub Action to auto format new PRs or just main or _something else_

Feel free to "Close as not planned" if this isn't important.

**Comment by jcbhmr** [comment]
Consider copying what the official devcontainers/features repo does! https://github.com/devcontainers/features/tree/main/.github/workflows

They have a lint and format workflow Action that uses https://github.com/azohra/shell-linter#readme
```yaml
- name: Shell Linter
  uses: azohra/shell-linter@v0.6.0
  with:
    path: "src/**/*.sh"
    severity: "error" # [style, info, warning, error]
  env:
    SHELLCHECK_OPTS: -e SC2072 # Acceptable use of decimal comparison
```
https://github.com/devcontainers/features/blob/main/.github/workflows/linter-automated.yaml#L14:L20

Sidenote: Maybe we should adopt some of the other workflows from the official devcontainers/features repo?

**Comment by danielbraun89** [comment]
Was easy enough to add shell formatting to generator using shfmt (prettier plugin is using this on the backend), and adding the github action you linked to. 
...and it had already catched a seemingly important error:
![image](https://user-images.githubusercontent.com/17760492/204154400-4ecd689e-1146-474d-9c22-f7cb56d48db0.png)

ill separate the styleguide into a different issue as thats quite more effort for me

**Comment by danielbraun89** [comment]
fixed by #92

**Comment by jcbhmr** [comment]
> ill separate the styleguide into a different issue as thats quite more effort for me

I think what we got right now with the linter and formatter and such is plenty 👍. If you want to add more documentation or guidelines, that's perfectly OK, but I recommend against going all out like an [Airbnb styleguide](https://github.com/airbnb/javascript#readme) or a [PEP8 styleguide](https://peps.python.org/pep-0008/) 😉.
